documentation/kubernetes.md

```diff
@@ -1,18 +1,123 @@
 
+In addition to its availability as a .NET CLI tool, the `dotnet monitor` tool is available as a prebuilt Docker image that can be run in container runtimes and orchestrators, such as Kubernetes.
+
+For Dockerfiles and repository information, see [Running in Docker](./docker.md)
+
+## Example Deployment
+
+The following example demonstrates a deployment of the dotnet-monitor container image monitoring two application containers within the same pod.
+
+```yaml
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: deploy-exampleapp
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: exampleapp
+  template:
+    metadata:
+      labels:
+        app: exampleapp
+    spec:
+      restartPolicy: Always
+      containers:
+      - name: app1
+        image: mcr.microsoft.com/dotnet/samples:aspnetapp
+        imagePullPolicy: Always
+        env:
+        - name: ASPNETCORE_URLS
+          value: http://+:80
+        - name: DOTNET_DiagnosticPorts
+          value: /diag/port
+        volumeMounts:
+        - mountPath: /diag
+          name: diagvol
+        - mountPath: /dumps
+          name: dumpsvol
+        resources:
+          limits:
+            cpu: 250m
+            memory: 512Mi
+      - name: app2
+        image: mcr.microsoft.com/dotnet/samples:aspnetapp
+        imagePullPolicy: Always
+        env:
+        - name: ASPNETCORE_URLS
+          value: http://+:81
+        - name: DOTNET_DiagnosticPorts
+          value: /diag/port
+        volumeMounts:
+        - mountPath: /diag
+          name: diagvol
+        - mountPath: /dumps
+          name: dumpsvol
+        resources:
+          limits:
+            cpu: 250m
+            memory: 512Mi
+      - name: monitor
+        image: mcr.microsoft.com/dotnet/monitor
+        # DO NOT use the --no-auth argument for deployments in production; this argument is used for demonstration
+        # purposes only in this example. Please continue reading after this example for further details.
+        args: [ "--no-auth" ]
+        imagePullPolicy: Always
+        env:
+        - name: DOTNETMONITOR_DiagnosticPort__ConnectionMode
+          value: Listen
+        - name: DOTNETMONITOR_DiagnosticPort__EndpointName
+          value: /diag/port
+        - name: DOTNETMONITOR_Storage__DumpTempFolder
+          value: /dumps
+        # DO NOT remove HTTPS from the URL for deployments in production; this is done for demonstration
+        # purposes only in this example. Please continue reading after this example for further details.
+        - name: DOTNETMONITOR_Urls
+          value: http://localhost:52323
+        volumeMounts:
+        - mountPath: /diag
+          name: diagvol
+        - mountPath: /dumps
+          name: dumpsvol
+        resources:
+          requests:
+            cpu: 50m
+            memory: 32Mi
+          limits:
+            cpu: 250m
+            memory: 256Mi
+      volumes:
+      - name: diagvol
+        emptyDir: {}
+      - name: dumpsvol
```

Comment:
This might be a dumb question since I've always done it this way and that may be because it doesn't work otherwise but...
Why don't we suggest having a `/diag` volume and then write dumps to `/diag/dumps` and use the diag port at `/diag/port` then the user only needs to define a single mount point for everything.

Reply:
Not a dumb question. I think that should be fine. Since these are all virtual volumes then it probably doesn't matter how they are split up regarding throughput. I'll give it a quick check and update the example as you suggested.